### ISSUE TITLE
Revamp to allow images with different aspect ratios

### ISSUE THREAD

**Issue by jrosindell** [OneZoom/OZtree#918]
We've got a design around square images which is nice but to what extent is this a pinch point - perhaps we should allow uncropped images to appear at least on leaves and possibly on signposts as well.

**Comment by davidebbo** [comment]
Sort of duplicate of https://github.com/OneZoom/tree-build/issues/78, though that one conflates ratings and crops.

**Comment by davidebbo** [comment]
I think we're reacting to seeing really bad square AI crops of skinny images. But things probably would look a lot more reasonable if we used the best possible square crops of those images.

Let's take a fairly extreme example: https://upload.wikimedia.org/wikipedia/commons/c/ce/Moanasaurus2.png

![image](https://github.com/user-attachments/assets/ca2f7c18-46a0-40be-873d-4afa08554624)

Right now, we're showing the tail:
![image](https://github.com/user-attachments/assets/ca15ae0c-53f2-41a7-ad98-0f5d6bde38e1)

But it could look like:

![image](https://github.com/user-attachments/assets/23b5fd67-e674-41e9-93ef-28380044fc82)

And of course, that doesn't capture the true shape of the animal, but it's not terrible, and it avoids all the issues of having a non-square image that ends up looking tiny. Sort of a case of finding the lesser evil...

**Comment by hyanwong** [comment]
> that doesn't capture the true shape of the animal, but it's not terrible

I agree. I think the best in this case would be a crop that allows a little whitespace above and below, but not the full length. That would require "more intelligent AI", but eventually will probably be workable. I have hand-cropped a number of the original OneZoom images rather like this. In this case that might include a bit of flipper, e.g. 
![image](https://github.com/user-attachments/assets/d1da79e9-0f82-4d1e-b29a-e1e3d89a530d)

**Comment by hyanwong** [comment]
The other thing we could do is to slightly bias the image choice in favour of aspect ratios that are slightly squarer, e.g. 
![this one](https://commons.wikimedia.org/wiki/File:Hainosaurus073.jpg) which would hopefully be cropped as:

<img width="274" alt="Screenshot 2024-11-26 at 09 53 21" src="https://github.com/user-attachments/assets/3023f9d9-fd17-4d11-934e-cc6e78eddec7">

**Comment by davidebbo** [comment]
That's for a different species, right? For https://en.wikipedia.org/wiki/Moanasaurus, it's just the one image.

The challenge is that it's hard to qualify the images. We do know that 'reconstructions' tend to be decent. If we replace an elongated reconstruction by a much more square image if a fossil fragment, it may not be an improvement.

> I think the best in this case would be a crop that allows a little whitespace above and below, but not the full length. That would require "more intelligent AI", but eventually will probably be workable.

Yes, I think this is a good direction. It could be mostly automated, where the only guidance we'd give it is left, right, up or down, and it would make a crop of that side, while guaranteeing to keep at least (say) 50% of the long side, by adding whitespace as needed. That might be a be cheaper than an all-out hand crafted selection (although with good tools, it's probably not hard either, especially if we all pitch in).
